### PR TITLE
feature-gcltocpu

### PR DESCRIPTION
**Pull request by psy3727**
-add new serverConfig settings gclToCPU and maxCPU
-add new config.utils methods getCPULimit, setCPULimit, enableGCLToCPU, disableGCLToCPU
-create new cron job gcltocpu

-add in configureable settings for baseCPU and stepCPU
-move all configuration settings from env.keys to config.utils

-update README.md
-change var and function names to approved AGS131 STYLE GUIDE v2178.328
-refactor functions to use async / await
-refactor cronjob to use destructuring

### CODE REVIEW

File: lib/common.js [hunk 1]
```diff
@@ -207,8 +207,56 @@ module.exports = (config) => {
...
+        }
       } catch (err) {
       }
+    },
+    async getCPULimit (username) {
+      if (!username) return 'Username required'
+      const usernameLower = username.toLowerCase()
+      const user = await db.users.findOne({ usernameLower })
+      if (!user) return 'Username not found'
+      return `CPU Limit is ${user.cpu} for user ${user.username}`
```
Comment (ags131) [inline lib/common.js]: Should `L` be capitalized?

File: lib/common.js [hunk 2]
```diff
@@ -207,8 +207,56 @@ module.exports = (config) => {
...
+    },
+    async setCPULimit (username, cpuLimit) {
+      if (!username) return 'Username required'
+      if (!cpuLimit || cpuLimit <= 0) return 'Please set a value > 0'
+      const usernameLower = username.toLowerCase()
+      const user = await db.users.findOne({ usernameLower })
+      if (!user) return 'Useranme not found'
```
Comment (ags131) [inline lib/common.js]: :stuck_out_tongue_winking_eye:

File: lib/common.js [hunk 3]
```diff
@@ -223,6 +271,11 @@ module.exports = (config) => {
   config.utils.setSocketUpdateRate._help = `setSocketUpdateRate(value) Sets socket update rate (in ms)`
   config.utils.getSocketUpdateRate._help = `getSocketUpdateRate() Returns current socket update rate`
   config.utils.setShardName._help = `setShardName(value) Sets the shard name`
+  config.utils.getCPULimit._help = `getCPULimit(username)`
+  config.utils.setCPULimit._help = `setCPULimit(username, value)`
+  config.utils.enableGCLToCPU._help = `enableGCLToCPU(maxCPU, baseCPU, stepCPU) changes running state and does not persist between server restarts`
```
Comment (ags131) [inline lib/common.js]: Help text should start with a capital letter, should also actually say what the function does (even if the function name should be self-explanatory, see set/getSocketUpdateRate and others above.)

File: lib/cronjobs/gcltocpu.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
+module.exports = (config) => {
+  config.cronjobs.gcltocpu = [10, async () => {
+    const { env, db } = config.common.storage
+    if (!db) return
+    const gclToCPU = config.utils.gclToCPU || false
```
Comment (ags131) [inline lib/cronjobs/gcltocpu.js]: Nitpick, but undefined is also falsy :stuck_out_tongue:

File: lib/common.js (hunk 1 above)
Comment (psy3727) [reply lib/common.js]: fixed

File: lib/common.js (hunk 3 above)
Comment (psy3727) [reply lib/common.js]: fixed

File: lib/cronjobs/gcltocpu.js (hunk 4 above)
Comment (psy3727) [reply lib/cronjobs/gcltocpu.js]: fixed

File: lib/common.js (hunk 2 above)
Comment (psy3727) [reply lib/common.js]: fixed

**Comment by psy3727** [comment]
npm test now comes back without errors

**Pull request merged**